src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -724,11 +735,19 @@ let run_scan_conf (caps : caps) (conf : Scan_CLI.conf) : Exit_code.t =
   let rules_and_origins =
     Lwt_platform.run (Lwt.pick (rules_and_origins :: spinner_ls))
   in
-
+  (match conf.engine_type with
+  | Engine_type.PRO { git_remote = Some url; _ } ->
+      let path = setup_repository url in
+      Sys.chdir (Fpath.to_string path)
```

Comment:
See the long comment, but I think that may solve the concerns I had about this being fragile by allowing it to be avoided. Otherwise I am potentially concerned about other components which rely on the current directory affecting scan settings.

Reply:
For posterity: Yea ended up going with hooks + project root, no need for change dir